js_modules/dagster-ui/packages/ui-core/src/runs/HourlyDataCache/HourlyDataCache.tsx

```diff
@@ -9,16 +9,24 @@ type Subscription<T> = (data: T[]) => void;
 export const defaultOptions = {
   expiry: new Date('3000-01-01'), // never expire,
 };
+
 export class HourlyDataCache<T> {
   private cache: Map<number, Array<TimeWindow<T>>> = new Map();
   private subscriptions: Array<{hour: number; callback: Subscription<T>}> = [];
-  private indexedDBCache?: ReturnType<typeof cache<'hourlyData', typeof this.cache>>;
+  private indexedDBCache?: ReturnType<typeof cache<string, typeof this.cache>>;
+  private indexedDBKey: string;
+
+  /**
+   * @param id A unique ID for the hourly data cache in this deployment
+   * @param [keyPrefix=''] A unique key identifying the timeline view [incorporating filters, etc.]
+   */
+  constructor(id?: string | false, keyPrefix = '') {
+    this.indexedDBKey = keyPrefix ? `${keyPrefix}-hourlyData` : 'hourlyData';
 
-  constructor(id?: string | false) {
     if (id) {
-      this.indexedDBCache = cache<'hourlyData', typeof this.cache>({
+      this.indexedDBCache = cache<string, typeof this.cache>({
         dbName: `HourlyDataCache:${id}`,
-        maxCount: 1, // We only store 1 entry
+        maxCount: 3, // >1 because we have both the overview timeline and backfill timelines
```

Comment:
I think we should have a separate LRU cache for the backfills page so that it doesn't evict the main Run Timeline cache